integration-cli/docker_cli_network_unix_test.go

```diff
@@ -257,6 +257,93 @@ func (s *DockerNetworkSuite) TestDockerNetworkLsDefault(c *check.C) {
 	}
 }
 
+func (s *DockerNetworkSuite) TestDockerNetworkLsNoPredefined(c *check.C) {
+	dockerCmd(c, "network", "create", "dev")
+	defer func() {
+		dockerCmd(c, "network", "rm", "dev")
+	}()
+
+	out, _ := dockerCmd(c, "network", "ls", "-f", "type=custom")
+	lines := strings.Split(out, "\n")
+	var preDefExists, devExists bool
+	for _, line := range lines[1 : len(lines)-1] {
+		netName := strings.Fields(line)[1]
+		if netName == "dev" {
+			devExists = true
+		} else if runconfig.IsPreDefinedNetwork(netName) {
+			preDefExists = true
+		}
+	}
+
+	// "dev" network should exists
+	c.Assert(devExists, checker.True, check.Commentf("out: %s", out))
+	// with filter 'type=custom', 'network ls' shouldn't display
+	// predefined network (bridge, host and none)
+	c.Assert(preDefExists, checker.False, check.Commentf("out: %s", out))
+}
+
+func (s *DockerNetworkSuite) TestDockerNetworkLsOnlyPredefined(c *check.C) {
+	dockerCmd(c, "network", "create", "dev")
+	defer func() {
+		dockerCmd(c, "network", "rm", "dev")
+	}()
+
+	out, _ := dockerCmd(c, "network", "ls", "-f", "type=builtin")
+	lines := strings.Split(out, "\n")
+	var hostExists, noneExists, bridgeExists, devExists bool
+	for _, line := range lines[1 : len(lines)-1] {
+		netName := strings.Fields(line)[1]
+		if netName == "dev" {
+			devExists = true
+		} else if runconfig.NetworkMode(netName).IsBridge() {
+			bridgeExists = true
+		} else if runconfig.NetworkMode(netName).IsHost() {
+			hostExists = true
+		} else if runconfig.NetworkMode(netName).IsNone() {
+			noneExists = true
+		}
+	}
+
+	// "dev" network shouldn't exists
+	c.Assert(devExists, checker.False, check.Commentf("out: %s", out))
+	// with filter 'type=builtin', 'network ls' should display
+	// predefined network (bridge, host and none)
+	c.Assert(bridgeExists && hostExists && noneExists, checker.True, check.Commentf("out: %s", out))
+}
+
+// filter with ID and name
+// only show 'bridge' and 'dev' network
+func (s *DockerNetworkSuite) TestDockerNetworkLsFilterByIDAndName(c *check.C) {
+	out, _ := dockerCmd(c, "network", "create", "dev")
+	defer func() {
+		dockerCmd(c, "network", "rm", "dev")
+	}()
+	containerID := strings.TrimSpace(out)
+
+	out, _ = dockerCmd(c, "network", "ls", "-f", "id="+containerID, "-f", "name=bridge")
+	lines := strings.Split(out, "\n")
+	var hostExists, noneExists, bridgeExists, devExists bool
+	for _, line := range lines[1 : len(lines)-1] {
+		netName := strings.Fields(line)[1]
+		if netName == "dev" {
```

Comment:
Also, this is duplicated from above as well, might be good to extract into a helper.